Robot_Adapter/Read/Elements/Node.cs

```diff
@@ -33,14 +34,15 @@ public partial class RobotAdapter
         /****           Private Methods                 ****/
         /***************************************************/
 
-        private List<Node> ReadNodes(List<string> ids = null)
+        private List<Node> ReadNodes(IList ids = null)
         {
+            List<int> nodeIds = CheckAndGetIds(ids);
             IRobotCollection robotNodes = m_RobotApplication.Project.Structure.Nodes.GetAll();
```

Comment:
WIth this change, would also move this line `IRobotCollection robotNodes = m_RobotApplication.Project.Structure.Nodes.GetAll();` into the first If-statement, as not needed if indices are provided.